pkg/reconciler/revision/resources/deploy.go

```diff
@@ -150,6 +157,7 @@ func rewriteUserProbe(p *corev1.Probe, userPort int) {
 
 func makePodSpec(rev *v1.Revision, cfg *config.Config) (*corev1.PodSpec, error) {
 	queueContainer, err := makeQueueContainer(rev, cfg)
+	tokenVolume := varTokenVolume.DeepCopy()
```

Comment:
Do we need to copy here?

Reply:
We must have deepCopy here to make makePodSpec idempotent. Otherwise, every call will try to add additional volume projections to the ones added in previous calls.  
We also use it in testing where it will fail since we call this function more than once.

Reply:
I wonder if we should stop keeping `varTokenVolume` as a constant, and construct it from a function.